src/spatio_temporal_voxel_layer.cpp

```diff
@@ -66,12 +66,12 @@ SpatioTemporalVoxelLayer::~SpatioTemporalVoxelLayer(void)
 void SpatioTemporalVoxelLayer::onInitialize(void)
 /*****************************************************************************/
 {
-  RCLCPP_INFO(node_->get_logger(),
+  RCLCPP_INFO((node_.lock())->get_logger(),
```

Comment:
At the start of this function you should `auto node = node_lock()` and check if its valid rather than continuously locking the weak pointer, that's not good behavior.